Notes/ContentViewModel.swift

```diff
@@ -0,0 +1,57 @@
+//
+//  ContentViewModel.swift
+//  Notes
+//
+//  Created by Lena Vadakkel on 03.11.23.
+//
+
+import Foundation
+import SwiftUI
+
+final class ContentViewModel: ObservableObject {
+    @Published var usernameInput: String = ""
+    @Published var passwordInput: String = ""
+    @Published var errorMessage = ""
+    @Published var usernameInvalid = false
+    @Published var passwordInvalid = false
+    @Published var isLinkActive = false
+        
+	enum ScreenEvent {
+		case signIn
+	}
+	
+	func onScreenEvent(_ event: ScreenEvent) {
+		switch event {
+			case .signIn:
+				usernameEqualToInput()
+		}
+	}
+	
+	
+    func usernameEqualToInput() {
+        let user = fetchUserByUsername(inputUsername: usernameInput)
```

Comment:
I tried to check both of them in one NSPredicate but it seems the NSPredicate only takes one value. 
If you meant to do one  NSPredicate for each I can't do that because the password is not unique (I'm not checking the uniqueness when the user is created, so that means it could be that two or more have the same password) but the username is that's why I'm checking the password afterwards.